tensorflow/compiler/tf2tensorrt/segment/segment.cc

```diff
@@ -1141,6 +1143,76 @@ Status SegmentGraph(const Graph* tf_graph,
       continue;
     }
     segments->emplace_back(itr.second.property, segment_nodes);
+    effective_nodes_counts.push_back(num_effective_nodes);
+  }
+
+  // --------------------------------- Step 4 ---------------------------------
+  // If the number of segments exceeds max_engines, prune the smallest ones.
+
+  int64_t max_trt_engine_ops;
+  TF_CHECK_OK(ReadInt64FromEnvVar("TF_TRT_MAX_ALLOWED_ENGINES",
+                                  /*default_value=*/20, &max_trt_engine_ops));
+
+  if (max_trt_engine_ops <= 0) {
+    LOG(WARNING) << "The environment variable TF_TRT_MAX_ALLOWED_ENGINES is "
+                 << "<= 0. TF-TRT did not limit the number of TensorRT engines "
+                 << "created.";
+
+  } else {
+    if (segments->size() > max_trt_engine_ops) {
+      LOG(WARNING) << "A total of " << segments->size() << " segments with at "
+                   << "least minimum_segment_size="
+                   << options.minimum_segment_size << " nodes have been found. "
+                   << "TF-TRT will only convert the " << max_trt_engine_ops
+                   << " largest segments. You can change this behavior by "
+                   << "modifying the environment variable "
+                   << "TF_TRT_MAX_ALLOWED_ENGINES=" << max_trt_engine_ops;
+
+      // Stable sort of the segment indices according to their effective sizes.
+      std::vector<int> indices(segments->size());
+      std::iota(indices.begin(), indices.end(), 0);
+
+      std::stable_sort(indices.begin(), indices.end(),
+                      [&effective_nodes_counts](int i1, int i2) {
+                        return effective_nodes_counts[i1] >
+                                effective_nodes_counts[i2];
+                      });
+
+      // Log which segments have been accepted or rejected
+      VLOG(1) << "The following segments have been accepted by TF-TRT:";
+      for (int i = 0; i < indices.size() && i < max_trt_engine_ops; i++) {
```

Comment:
Nitpick: `indices.size() = segments->size()` and `segments->size() > max_trt_engine_ops` so this check is redundant, `i < max_trt_engine_ops` is enough.

Reply:
Fixed as merged with the other loop